Source/JavaScriptCore/runtime/ISO8601.cpp

```diff
@@ -1599,6 +1599,52 @@ Int128 ExactTime::round(Int128 quantity, unsigned increment, TemporalUnit unit,
     return roundNumberToIncrement(quantity, incrementNs, roundingMode);
 }
 
+// https://tc39.es/proposal-temporal/#sec-temporal-datedurationsign
+static int32_t dateDurationSign(const Duration& d)
+{
+    if (d.years() > 0)
+        return 1;
+    if (d.years() < 0)
+        return -1;
+    if (d.months() > 0)
+        return 1;
+    if (d.months() < 0)
+        return -1;
+    if (d.weeks() > 0)
+        return 1;
+    if (d.weeks() < 0)
+        return -1;
+    if (d.days() > 0)
+        return 1;
+    if (d.days() < 0)
+        return -1;
+    return 0;
+}
+
+// https://tc39.es/proposal-temporal/#sec-temporal-internaldurationsign
+int32_t ISO8601::InternalDuration::sign() const
+{
+    int32_t sign = dateDurationSign(m_dateDuration);
+    if (sign)
+        return sign;
+    return timeDurationSign();
+}
+
+// https://tc39.es/proposal-temporal/#sec-temporal-combinedateandtimeduration
+InternalDuration InternalDuration::combineDateAndTimeDuration(JSGlobalObject* globalObject, Duration dateDuration, Int128 timeDuration)
+{
+    VM& vm = globalObject->vm();
+    auto scope = DECLARE_THROW_SCOPE(vm);
+
+    int32_t dateSign = dateDurationSign(dateDuration);
+    int32_t timeSign = timeDuration < 0 ? -1 : timeDuration > 0 ? 1 : 0;
+    if (dateSign && timeSign && dateSign != timeSign) {
+        throwRangeError(globalObject, scope, "Date and time have different signs"_s);
+        return { };
+    }
```

Comment:
In the spec text, this is an assertion. (I think I remember changing this at some point, maybe around last October.) Is it feasible to make it an assertion here as well? In the spec, there's only one caller of this operation where we could end up with mismatched signs, and it is checked explicitly before calling CombineDateAndTimeDuration. At all other call sites, the signs can be trivially deduced to be either equal or zero.

This would allow removing the `globalObject` parameter.

Reply:
Fixed in [a86202d](https://github.com/WebKit/WebKit/pull/43980/commits/a86202d7ed692fdb479134c1ac06cfa9399a893f) (the only reason I didn't do it that way in the first place is that I wrote this code before the spec was changed, most likely).